presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/WorkerProperty.java

```diff
@@ -22,17 +22,58 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
-/**
- * A utility class that helps with properties and its materialization.
- */
-public class WorkerProperty
+public abstract class WorkerProperty<T1 extends NativeExecutionConnectorConfig, T2 extends NativeExecutionNodeConfig, T3 extends NativeExecutionSystemConfig>
```

Comment:
Why do we need generics here?

Reply:
Another question:
Are we able to make WorkerProperty not an abstract class and combine contents from PrestoSparkWorkerProperty in here?

Reply:
Thanks for explaining. We can let getSystemConfig() return SystemConfig. 
So FacebookSystemConfig is not a parent-child relationship with SystemConfig. So FacebookWorkerProperty should have both FacebookSystemConfig and SystemConfig(which is in its parent WorkerProperty):
```
WorkerProperty
 - SystemConfig
 + SystemConfig getSystemConfig() {...} 

FacebookWorkerProperty extends WorkerProperty
 - FacebookSystemConfig
```

We can chat offline if it is unclear.